test/integration/rewrite-with-browser-history/next.config.js

```diff
@@ -0,0 +1,10 @@
+module.exports = {
+  rewrites() {
+    return [
+      {
+        source: '/:pagePrefix/:path*',
+        destination: '/dynamic-page/:path*',
```

Comment:
This is where it goes wrong I guess, the first param is not defined in the second path and the `[[...param]]` then doesn't get defined and the router thinks it can't match the route.